cupy/cuda/cub.pyx

```diff
@@ -572,3 +573,15 @@ cpdef cub_scan(_ndarray_base arr, op):
         return device_scan(arr, op)
 
     return None
+
+
+cpdef cub_histogram(_ndarray_base x, _ndarray_base y, bins):
+    """Check if the required workspace size is too much, if not then proceed
+    to compute the histogram, otherwise return None. This is a workaround for
+    NVIDIA/cub#613.
+    """
+    try:
+        out = device_histogram(x, y, bins)
+    except _memory.OutOfMemoryError:
```

Comment:
Would it be possible to set a threshold value based on `bins` rather than just make this error?

Reply:
It's possible, but how much threshold is needed?

Reply:
(but it'd only reduce the chance of hitting OOM, not eliminate it)

Reply:
The threshold of `bins` will change based on the device's memory\capability making such a solution device-specific.

Reply:
I'm wondering whether we should disable CUB in `cupy.histogram` until CUB issue gets fixed. It is an unpredictable behavior for users to use so much memory even if it fits in the memory.

Reply:
This is a corner case in which users have huge number of bins (I have no idea why @ogreen did that TBH 😂), but for other use cases, CUB is still usable and preferred I'd say.

Reply:
I think that so long as the application doesn't crash, then based on what I have seen CUB's implementation is very efficient. However, it would be nice to have a fallback in case it fails.
FWIW, the application that I was using histogram was in the context of graphs. I was attempting to get the adjacency list size for each vertex in the graph. While this can be accomplished with atomic operations, it would seem redundant to implement it as it is exactly what a histogram does.